### ISSUE TITLE
Unsplash support?

### ISSUE THREAD

**Issue by gigiboeru** [Owyn/HandyImage#527]
Can you please add Unsplash? I tried adding the website on the User includes & User matches but I'm doing something wrong. 

For instance https://images.unsplash.com/photo-1553969420-fb915228af51?ixid=MnwxMTI1OHwwfDF8cmFuZG9tfHx8fHx8fHx8MTY3OTAwNTM1OA&ixlib=rb-4.0.3&q=85&w=1920 to be rewritten to https://images.unsplash.com/photo-1553969420-fb915228af51 (the latter is the high res no re-compression url).

**Comment by Owyn** [comment]
user include works just fine for me?
![image](https://user-images.githubusercontent.com/1309656/225864827-a3331517-fa40-478c-bf9e-c6d81cf496bc.png)

**Comment by gigiboeru** [comment]
Thanks for the quick reply, but that doesn't work as I want it. My aim is to get the full resolution uncompressed free picture. Hope I make sense.  

For instance https://unsplash.com/photos/PPuIEscvPfM does open a pic but it's .avif  - what I want is https://images.unsplash.com/photo-1678876403327-13ac6aa6e8aa (I get this in FireFox by using "Open image in new tab" and manually removing anything after the question mark https://images.unsplash.com/photo-1678876403327-13ac6aa6e8aa?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1170&q=80 - basically format, resolution, quality) So the unsplash.com/photos/ID changes to images.unsplash.com/photo-ID.

**Comment by Owyn** [comment]
There's also this button
![image](https://user-images.githubusercontent.com/1309656/225913375-9d90be38-bab9-474f-afce-14be61ceb99e.png)

**Comment by Owyn** [comment]
Also in the gallery there's a shortcut for that button
![image](https://user-images.githubusercontent.com/1309656/225955052-7f9123fd-c43b-4ce3-b6ac-386030a5eb37.png)

Do you want the script to just auto-press that button for every image when visiting their page? cuz that's what it gonna do to get the original image

**Comment by gigiboeru** [comment]
If possible, I want the script to parse the the URL and rewrite it to get that full/largest image.

I'm not using the Unsplash website to get the images, but an extension named Tabliss (from any new tab I just right click and choose "open image in new tab"). 

I was wondering if handyimage can rewrite the URL to a simpler form (which will grant access to the original image).

**Comment by Owyn** [comment]
> If possible, I want the script to parse the the URL and rewrite it to get that full/largest image.

that download original button seems to have the full\largest image, 
and the displayed image (with cut url) is a re-encode of it as far as I've tested

are you not opening these tabs from a gallery page? cuz there's a download button there you could press instead of opening a tab 🤔  (my prev screen)

> I was wondering if handyimage can rewrite the URL to a simpler form (which will grant access to the original image).

yea, with "open images directly with browser" option in the script, that seems to be its main use - to let ppl use tab downloader

**Comment by gigiboeru** [comment]
> are you not opening these tabs from a gallery page?

No, I open them from an extension, by using right click and "open image in new tab"


>  "open images directly with browser"

When I press the "Configuration" button nothing happens.

**Comment by Owyn** [comment]
> No, I open them from an extension, by using right click and "open image in new tab"

I mean the pages with images, - where you find those images in the first place, before they come into your extension.
cuz there's a `download` button there one can use instead of opening tabs:
![image](https://user-images.githubusercontent.com/1309656/226986393-61c3bea2-02e6-4d4a-b899-7a3d213b0966.png)

> When I press the "Configuration" button nothing happens.

you have to press it when the extension is active and is displaying a fullsized image
